### PR TITLE
Style headerlinks for table captions properly

### LINKED ISSUE #437: Tables defined using table directive gets visible permalink on table title
When using the table directive with a table title, the headerlink character is always visible.
For normal headers the symbol is only visible when hovering with the mouse pointer.

```
Escape sequences
----------------

.. table:: Escape sequences
   :widths: auto

   ================ ==============
   Escape sequence  Value
   ================ ==============
   \\n              new-line
   \\t              tab
   \\v              vertical tab
   \\b              backspace
   \\r              carriage return
   \\f              form feed
   \\a              alert
   \\\ \\           backslash
   \\?              question mark
   \\'              single quote
   \\"              double quote
   \\0              null character
   \\ooo            octal
   \\xhhh           hexadecimal
   ================ ==============
```

image

### PR DESCRIPTION
**Pull request by mitya57**
Fixes #437.

Now headerlinks for tables behave the same way like for headers: invisible by default, showing on hover, using Font Awesome.

### CODE REVIEW

**Comment by mitya57** [comment]
Ping?

**Comment by Blendify** [comment]
I am doing a dev sprint this weekend will take a look then.

**Comment by mitya57** [comment]
Great, thanks. Please also look at #432 if possible.

**Comment by Blendify** [comment]
The changes are ok but not perfect, when you hover over the text it shift to the left. Can you try to fix this?

**Comment by mitya57** [comment]
@Blendify Please test the latest version.

**Comment by mitya57** [comment]
Ping?

**Comment by ericholscher** [comment]
Looks like a reasonable change, is there a good example (gif?) of the new change.

**Comment by benjaoming** [comment]
@Blendify
> The changes are ok but not perfect

Subsequent PRs can amend what's an overall improvement

@mitya57 could you post a screenshot to convince the maintainers to press the "merge" button on this and #432 ? :)

**Comment by Blendify** [comment]
I will look at current PRs this weekend

**Comment by mitya57** [comment]
Here is a screenshot of what changed.

before
after

Previously the ¶ symbol was always visible, now it is only visible on hover.

File: sass/_theme_rst.sass [hunk 1]
```diff
@@ -106,19 +106,16 @@
...
       @extend .fa
       &:after
-        visibility: visible
         content: "\f0c1"
         font-family: FontAwesome
-        display: inline-block
-    &:hover .headerlink
-      display: inline-block
+    &:hover .headerlink:after
+      visibility: visible
```
Comment (jessetan) [inline sass/_theme_rst.sass]: LGTM, although I think slightly reducing the link for table captions looks nicer, by adding:
``` sass
  table > caption .headerlink:after
    font-size: 12px
```
Current PR: 
14px

With `font-size: 12px`:
12px

Comment (mitya57) [reply sass/_theme_rst.sass]: Added!

**Pull request merged**